### PR TITLE
Split JS files into one file per partial

### PR DESCRIPTION
**Pull request by gkrishnan724**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [1] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [1] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by Siddhartha1234** [comment]
@gkrishnan724 Please rebase the commits into a single commit (squash) before it can be merged.

File: index.html [hunk 1]
```diff
@@ -9,13 +9,24 @@
...
+    <script type="text/javascript" src="resources/js/routeprovider.js"></script>
+    <script type="text/javascript" src="resources/js/Controllers/SnippetsController.js"></script>
+    <script type="text/javascript" src="resources/js/Controllers/TabController.js"></script>
+    <script type="text/javascript" src="resources/js/Directives/HomeDirective.js"></script>
+    <script type="text/javascript" src="resources/js/Directives/LanguageDirective.js"></script>
+    <script type="text/javascript" src="resources/js/Directives/TryonlineDirective.js"></script>
+    <script type="text/javascript" src="resources/js/filters/format_descFilter.js"></script>
+    <script type="text/javascript" src="resources/js/filters/toArrayFilter.js"></script>
+    <script type="text/javascript" src="resources/js/filters/orderEmptyfilter.js"></script>
+    <script type="text/javascript" src="resources/js/Directives/getinvolvedDirective.js"></script>
```
Comment (hemangsk) [inline index.html]: @gkrishnan724 Won't this decrease the site's performance because of a large number of files being loaded? I really think that it will affect the speed

File: resources/js/filters/toArrayFilter.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+app.filter("toArray", function(){
```
Comment (hemangsk) [inline resources/js/filters/toArrayFilter.js]: I think we can have all the filters in a single file, will decrease the num of files being loaded

File: resources/js/config.js [hunk 3]
```diff
@@ -1 +1 @@
-api_link = "https://webservices.coala.io"
+api_link = "https://webservices.coala.io"
```
Comment (hemangsk) [inline resources/js/config.js]: Please insert a new line here 😃

File: resources/js/filters/format_descFilter.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        if (lastspace != -1) {
+            if (value.charAt(lastspace-1) == ',') {
+                lastspace = lastspace - 1;
+            }
+            value = value.substr(0, lastspace);
+        }
+
+        return value;
+    };
+});
```
Comment (Nosferatul) [inline resources/js/filters/format_descFilter.js]: New line here too.

**Comment by hemangsk** [comment]
@gkrishnan724 Home-Directive is still in app.js, so you'll have to remove it from there to avoid collision and duplication.

check out the related console error which comes on page load,
```
Multiple directives [home (module: coala), home (module: coala)] asking for template on: <home class="ng-scope">
```

File: index.html [hunk 5]
```diff
@@ -31,6 +36,7 @@
     <script type="text/javascript" src="components/ngstorage/ngStorage.min.js"></script>
 
 
+
```
Comment (yukiisbored) [inline index.html]: Why is this needed?

**Comment by gkrishnan724** [comment]
Sorry about the silly mistakes :/ please review it now :)

**Comment by gkrishnan724** [comment]
Sorry about the silly mistakes :/ please review it now :)

**Comment by yukiisbored** [comment]
@hemangsk Please do a review. I can't review JS code, these callbacks is making my head dizzy.

**Comment by hemangsk** [comment]
@yukiisbored @gkrishnan724  Yeah I'm testing it locally 😃  looks all awesome so far

**Comment by hemangsk** [comment]
ack cadc854

**Comment by hemangsk** [comment]
@rultor merge
